mozilla-release/mobile/android/base/java/org/mozilla/gecko/BrowserApp.java

```diff
@@ -3403,7 +3407,11 @@ private void showHomePagerWithAnimator(String panelId, Bundle panelRestoreData,
         // This must be called before the dynamic toolbar is set visible because it calls
         // FormAssistPopup.onMetricsChanged, which queues a runnable that undoes the effect of hide.
         // With hide first, onMetricsChanged will return early instead.
-        mFormAssistPopup.hide();
+        /* Cliqz Start */
+        if (mFormAssistPopup != null) {
```

Comment:
mFormAssistPopup shouldn't be null, except when onDestroy called on GeckoApp
why we need this check ?

Reply:
Because it was throwing `NullPointerException`. But I can't remember how to reproduce this crash now.

Reply:
So please remove this line and ask javier for multiple scenarios to test the app on modified part related.
and if it crashs try to debug how it reach here with null, instead of overcome it by doing if condition